Add Governance (#709)
bambinos/bambi: GOVERNANCE.md

The Project is developed by a team of distributed developers, called Contributors. 
Contributors are individuals who have contributed code, documentation, designs or other work to one 
or more Project repositories. Anyone can be a Contributor. 
Contributors can be affiliated with any legal entity or none. 
Contributors participate in the project by submitting, reviewing and discussing GitHub Pull Requests
and Issues and participating in open and public Project discussions on GitHub and Slack. 
The foundation of Project participation is openness and transparency.

There have been many Contributors to the Project, whose contributions are listed in the logs of any 
of the repositories under the bambinos organization.

The Project Community consists of all Contributors and Users of the Project. 
Contributors work on behalf of and are responsible to the larger Project Community and we strive to 
keep the barrier between Contributors and Users as low as possible.

## Governance

### Community Architecture

* General Contributors
* Recurrent Contributors
* Core Contributors (of which Council members are also a part of)
* Steering Council

> Anyone working with Bambi has the responsibility to personally uphold the Code of Conduct. 

Reviewer: Link code of conduct

can collectively make decisions for the health of the project.

##### Length of Tenure and Reverification

<!-- NOTE Everything here is open to discussion -->

* Council members term limits are 4 years, after which point their seat will come up for reelection.
* Each year on April 7th council members will be asked to restate their commitment to being on the 
Council.
* Attempts should be made to reach every council member over at least 2 communication media. 
For example: email, Slack, phone, or GitHub.
* If a Council Member does not restate their commitment their seat will be vacated.
* Inactivity can be determined by lack of substantial contribution, including votes on council, 
code or discussion contributions, contributions in the community or otherwise.
* In the event of a vacancy in the council, an [election](#council-selection-process) will be held 
to fill the position.
* There is no limit on the number of terms a Council Member can serve

##### Current Council members

The current Council members are:

* Gabriel Stechschulte ([@gstechschulte](https://github.com/gstechschulte))
* Osvaldo Martin ([@aloctavodia](https://github.com/aloctavodia))

> * Ravin Kumar ([@canyon289](https://github.com/canyon289))

Reviewer: ```suggestion
```

Removing myself so you three have less voting overhead and with votes for people that are more familiar with the package

# Main Governance Document

## The Project

The Bambi Project (The Project) is an open source software project.

> The goal of The Project is to develop open source software and deploy open and public websites and 

Reviewer: ```suggestion
The goal of The Project is to develop open source software and deploy open and public content and 
```
No change needed. websites seems specific, perhaps this can be broadened

Contributors can be affiliated with any legal entity or none. 
Contributors participate in the project by submitting, reviewing and discussing GitHub Pull Requests
and Issues and participating in open and public Project discussions on GitHub and Slack. 
The foundation of Project participation is openness and transparency.

There have been many Contributors to the Project, whose contributions are listed in the logs of any 
of the repositories under the bambinos organization.

The Project Community consists of all Contributors and Users of the Project. 
Contributors work on behalf of and are responsible to the larger Project Community and we strive to 
keep the barrier between Contributors and Users as low as possible.

## Governance

### Community Architecture

* General Contributors
* Recurrent Contributors
* Current Core Contributors (of which Council members are also a part of)
* Steering council <!-- Do we specify the number of persons?-->

Anyone working with Bambi has the responsibility to personally uphold the Code of Conduct. 
Core Contributors have the additional responsibility of _enforcing_ the Code of Conduct to maintain
a safe community.

#### Recurrent Contributors

> Recurrent Contributors are those individuals who contribute recurrently to theproject and can 

Reviewer 2: ```suggestion
Recurrent Contributors are those individuals who contribute recurrently to the project and can 
```

> Recurrent Contributors are those individuals who contribute recurrently to theproject and can 

Reviewer 2: ```suggestion
Recurrent Contributors are those individuals who contribute recurrently to the project and can 
```

candidate -- "would I trust this person to lead Bambi?".
* Candidates are evaluated independently, each candidate having 60% or more of yes votes _and_ less 
or equal than 20% of no votes is chosen. If the number of chosen candidates is >=4 and <=10 all 
candidates are confirmed and the election process stops here.
* In the event that either not enough or too many candidates were confirmed, candidates are ranked 
by interpreting yes=+1, neutral=0 and no=-1. If too many candidates were confirmed, the 10 
candidates with higher rank are elected. If not enough candidates were chosen, the 4 candidates with
 higher rank are elected.
* In the event of a tie there will be a runoff election for the tied candidates. To avoid further 
ties and discriminate more among the tied candidates, this vote will be held by 
[Majority Judgment](https://en.wikipedia.org/wiki/Majority_judgment) (MJ): for each candidate, 
voters judge their suitability for office as either "Excellent", "Very Good", "Good", "Acceptable", 
"Poor", or "Reject". Multiple candidates may be given the same grade by a voter. 
The candidate with the highest median grade is the winner.

> * If more than one candidate has the same highest median-grade, the MJ winner isdiscovered by 

Reviewer 2: ```suggestion
* If more than one candidate has the same highest median-grade, the MJ winner is discovered by 
```

removing (one-by-one) any grades equal in value to the shared median grade from each tied 
candidate's total. This is repeated until only one of the previously tied candidates is currently 
found to have the highest median-grade.
* If ties are still present after this second round, the winner will be chosen at random. 
Each person tied will pick an integer number in the `[1, 100]` interval and send it privately to the
third party mediating the election. After receiving all the numbers, said third party will draw a 

> random integer from random.org. The person with theclosest circular distance, defined as 

Reviewer 2: ```suggestion
random integer from random.org. The person with the closest circular distance, defined as 
```

`min(|a-b|, 100-|a-b|)`, will be selected. This process will be repeated as many times as necessary
as there may be ties resulting from candidates choosing the same number.
* At the conclusion of voting, all the results will be posted. And at least 24 hours will be left to
challenge the election result in case there were suspicions of irregularities or the process had not

> been correctly carriedout.

Reviewer 2: ```suggestion
been correctly carried out.
```

* For the first election only the people registered following the guidelines in `elections/ArviZ_2020.md` can vote
* In the first year, the council must determine voting eligibility for future elections between two criteria:
  * Core contributors
  * The contributing community at large

## Historical note

Traditionally, project leadership was unstructured but primarily driven by a subset of 
Core Contributors whose active and consistent contributions have been recognized by their receiving 
"commit rights" to the Project GitHub repositories. 
In general all Project decisions are made through consensus among the Core Contributors with input 
from the Community.

While this approach has served us well, as the Project grows and faces more legal and financial 
decisions and interacts with other institutions, we see a need for a more formal governance model. 
Moving forward The Project leadership will consist of a **TODO NAME** Council. 
We view this governance model as the formalization of what we are already doing, rather than a 
change in direction.

<!-- NOTES:  

Include something about our relationship with PyMC (PyMC helped us by giving us access to GSoC three times)

> How do we start a Council if we don't have 'core' contributors yet?

Reviewer: Identify core contributors first through some critieria. Examples are asking anyone that's contributed to self nominate, that way you can start with folks who are involved and actively interested.
